lib/src/arithmetic/values/fixed_point_value.dart

```diff
@@ -0,0 +1,167 @@
+//
+// 2024 September 24
+// Authors:
+//  Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'dart:math';
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An immutable representation of (un)signed fixed-point values following
+/// Q notation (Qm.n format) as introduced by
+/// (Texas Instruments)[https://www.ti.com/lit/ug/spru565b/spru565b.pdf].
+@immutable
+class FixedPointValue implements Comparable<FixedPointValue> {
+  /// The full fixed point value bit storage in two's complement.
+  late final LogicValue value;
+
+  /// The sign of the fixed point number.
+  final LogicValue sign;
+
+  /// The integer part of the fixed-point number.
+  final LogicValue integer;
+
+  /// The fractional part of the fixed-point number.
+  final LogicValue fraction;
+
+  /// Constructs [FixedPointValue] from sign, integer and fraction values.
+  FixedPointValue(
+      {this.sign = LogicValue.empty,
+      this.integer = LogicValue.empty,
+      this.fraction = LogicValue.empty}) {
+    if ((integer == LogicValue.empty) & (fraction == LogicValue.empty)) {
+      throw RohdHclException('integer or fraction must be non-empty');
+    }
+    if ((sign == LogicValue.empty) | (sign.isZero)) {
+      value = [sign, integer, fraction].swizzle();
+    } else {
+      value = ~[LogicValue.zero, integer, fraction].swizzle() + 1;
+    }
+  }
+
+  /// Constructs [FixedPointValue] of a Dart [double] rounding away from zero.
+  factory FixedPointValue.ofDouble(double value,
+      {required bool signed, required int m, required int n}) {
+    if (value.abs().floor() > pow(2, m) - 1) {
+      throw RohdHclException('value exceed integer part');
+    }
+    if ((!signed) & (value < 0)) {
+      throw RohdHclException('Negative value must be signed.');
+    }
+    final sign = value >= 0 ? LogicValue.zero : LogicValue.one;
+    final integerPart = value.abs().floor();
+    final fractionalPart = ((value.abs() - integerPart) * pow(2, n)).round();
+
+    return FixedPointValue(
+        sign: signed ? sign : LogicValue.empty,
+        integer: LogicValue.ofInt(integerPart, m),
+        fraction: LogicValue.ofInt(fractionalPart, n));
+  }
+
+  /// Returns the value of the fixed-point number in a Dart [double] type.
+  double toDouble() {
+    final value = integer.toInt().toDouble() +
+        (fraction.toInt().toDouble() / pow(2, fraction.width));
+    return (sign == LogicValue.empty) | (sign == LogicValue.zero)
+        ? value
+        : -value;
+  }
+
+  /// Returns a negative integer if `this` less than [other],
+  /// a positive integer if `this` greater than [other],
+  /// and zero if `this` and [other] are equal.
+  @override
+  int compareTo(Object other) {
+    if (other is! FixedPointValue) {
+      throw RohdHclException('Input must be of type FixedPointValue');
+    }
+    final thisValue = toDouble();
+    final otherValue = other.toDouble();
+    if (thisValue == otherValue) {
```

Comment:
could do `toDouble().compareTo(other.toDouble()`?

Although, also, this only scales up to numbers that fit (with enough precision) in a Dart `double`.  Should we implement a comparison using `LogicValue` instead (and considering sign, fraction, etc.) so that it works for any `FixedPoint` instead of only ones reliably representable as a `double`?